toolchain/parse/node_ids.h

```diff
@@ -138,7 +138,10 @@ using AnyFunctionDeclId = NodeIdOneOf<FunctionDeclId, FunctionDefinitionStartId,
 using AnyImplDeclId = NodeIdOneOf<ImplDeclId, ImplDefinitionStartId>;
 using AnyInterfaceDeclId =
     NodeIdOneOf<InterfaceDeclId, InterfaceDefinitionStartId>;
-using AnyNamespaceId = NodeIdOneOf<NamespaceId, ImportDeclId>;
+using AnyNamespaceId =
+    NodeIdOneOf<NamespaceId, ImportDeclId, LibraryDeclId, PackageDeclId>;
+using AnyPackagingDeclId =
+    NodeIdOneOf<ImportDeclId, LibraryDeclId, PackageDeclId>;
```

Comment:
Does this work:
```suggestion
using AnyPackagingDeclId =
    NodeIdOneOf<ImportDeclId, LibraryDeclId, PackageDeclId>;
using AnyNamespaceId =
    NodeIdOneOf<NamespaceId, AnyPackagingDeclId>;
```
? No worries if not, but if this does work, that might allow your commented out `static_assert` to also work.

Reply:
Nope, I checked and it looks like this would require extra effort to support too. Let's not.

Reply:
I was mulling this, but probably best saving for a separate PR if we want to add support. :)